pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+                              'developers find \'constant\' works best for DEIMOS.'
+
+        defaults['pca'] = PCAPar()
+        dtypes['pca'] = [ ParSet, dict ]
+        descr['pca'] = 'Parameters used for the PCA parameters in slit tracing'
+
+        # Instantiate the parameter set
+        super(TraceSlitsPar, self).__init__(list(pars.keys()),
+                                            values=list(pars.values()),
+                                            defaults=list(defaults.values()),
+                                            options=list(options.values()),
+                                            dtypes=list(dtypes.values()),
+                                            descr=list(descr.values()))
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, cfg):
+        k = cfg.keys()
+        parkeys = [ 'function', 'polyorder', 'medrep', 'number', 'trim', 'maxgap', 'maxshift',
+                    'pad', 'sigdetect', 'fracignore', 'diffpolyorder', 'single', 'sobel_mode' ]
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+
+        pk = 'pca'
+        kwargs[pk] = PCAPar.from_dict(cfg[pk]) if pk in k else None
+
+        return cls(**kwargs)
+
+    @staticmethod
+    def valid_functions():
+        """
+        Return the list of valid functions to use for slit tracing.
+        """
+        return [ 'polynomial', 'legendre', 'chebyshev' ]
+
+    @staticmethod
+    def valid_sobel_modes():
+        """Return the valid sobel modes."""
+        return [ 'nearest', 'constant' ]
+
+    def validate(self):
+        if self.data['number'] == 0:
+            raise ValueError('Number of slits must be -1 for automatic identification or '
+                             'greater than 0')
+        
+class TraceTiltsPar(ParSet):
+    """
+    Parameters specific to PypIts slit tracing algorithm
+    """
+    def __init__(self, idsonly=None, method=None, params=None, order=None, disporder=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])      # "1:" to skip 'self'
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+
+        defaults['idsonly'] = False
+        dtypes['idsonly'] = bool
+        descr['idsonly'] = 'Only use the arc lines that have an identified wavelength to trace ' \
+                           'tilts'
+
+        defaults['method'] = 'spline'
```

Comment:
The spline and PCA have been turned off.

Reply:
Okay, leaving it in.